### PR TITLE
Search within a range of dates for creation/moved date field

### PR DESCRIPTION
**Pull request by ksciana**
Solution for #3942

### CODE REVIEW

File: app/Filter/BaseDateRangeFilter.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     * @access protected
+     * @param  string $field
+     */
+    protected function applyDateFilter($field)
+    {
+        $dates = explode('..', $this->value);
+        
+        if(count($dates)=== 2){          
+            $timestampFrom = $this->dateParser->getTimestampFromIsoFormat($dates[0]);
+            $timestampTo = $this->dateParser->getTimestampFromIsoFormat($dates[1]);
```
Comment (fguillot) [inline app/Filter/BaseDateRangeFilter.php]: To support user defined date format and ISO format, you should use the method `getTimestamp()` from the `DateParser` class instead.

In this way, `createdRange:"2017-12-09..2017-12-15"` and `createdRange:"09/12/2017..15/12/2017"` should work.

Comment (ksciana) [reply app/Filter/BaseDateRangeFilter.php]: OK, but in ``BaseDateFilter`` class is used ``getTimestampFromIsoFormat()`` method form the ``DateParser`` class. It was reference for me.

**Comment by ksciana** [comment]
New changes waiting for review.

**Comment by letompouce** [comment]
@ksciana out of curiosity, why didn't you extend this to the Due date as well?

**Pull request merged**